src/core/templates/show-logo.js

```diff
@@ -1,35 +1,45 @@
 // @ts-check
+import { docLink, showError } from "../../core/utils.js";
 import { html } from "../../core/import-maps.js";
-import { showWarning } from "../../core/utils.js";
 
 const name = "core/templates/show-logo";
 
 /**
+ * Logo mapper. Takes a logo structure and converts it to HTML.
+ *
  * @param {object} logo
  * @param {string} logo.src
  * @param {string} logo.url
  * @param {string} logo.alt
  * @param {string} [logo.id]
  * @param {number} [logo.width]
  * @param {number} [logo.height]
+ * @param {number} index
  */
-export default function showLogo(logo) {
+export default function showLogo(logo, index) {
   /** @type {HTMLAnchorElement} */
-  const a = html`<a href="${logo.url || ""}" class="logo"></a>`;
-  if (!logo.alt) {
-    const msg = "Found spec logo without an `alt` attribute.";
-    showWarning(msg, name, { elements: [a] });
+  const a = html`<a href="${logo.url ?? null}" class="logo"
+    ><img
+      id="${logo.id ?? null}"
+      alt="${logo.alt ?? null}"
+      width="${logo.width ?? null}"
+      height="${logo.height ?? null}"
```

Comment:
The `?? null` are unnecessary here. If value is null or undefined (which is what `??` will check for), hyperHTML will not set the attribute (unless you're being explicit here).

Reply:
ah, I wanted to assure that empty strings also got treated as `null`.

Reply:
Current code produces with empty strings, for example:

```
<img id="" alt="" width="0" height="0" src="">
```

With the `null`s, the attributes won't be there, which is more what I expect.

Reply:
Actually, you are right... here I really want ` || null`,  not `?? null`.